src/diff.rs

```diff
@@ -61,11 +61,75 @@ pub struct ModifiedFile {
 
 #[derive(Debug, PartialEq, Eq)]
 pub struct FileDiff {
-    // TODO
+    pub hunks: Vec<Hunk>,
+}
+
+#[derive(Debug, PartialEq, Eq)]
+pub struct Hunk {
+    pub header: Line,
+    pub lines: Vec<LineDiff>,
+}
+
+/// The content of a single line.
+pub type Line = Vec<u8>;
```

Comment:
I wonder if we should use a ref instead of an owned type here. Less copying for large diffs, but maybe more of a pain to use. Though this is how `git2` works.

Reply:
The proxy would have to do a clone anyway to send it to the front end right?

Reply:
Yeah true